src/Microsoft.OData.Core/ODataMetadataOutputContext.cs

```diff
@@ -57,6 +72,55 @@ internal sealed class ODataMetadataOutputContext : ODataOutputContext
             }
         }
 
+        /// <summary>
+        /// Asynchronously flush the underlying stream.
+        /// </summary>
+        /// <returns></returns>
+        internal Task FlushAsync()
+        {
+            this.AssertAsynchronous();
+
+            return this.xmlWriter.FlushAsync();
+        }
+
+        /// <summary>
+        /// Writes an <see cref="ODataError"/> into the message payload.
+        /// </summary>
+        /// <param name="error">The error to write.</param>
+        /// <param name="includeDebugInformation">
+        /// A flag indicating whether debug information (e.g., the inner error from the <paramref name="error"/>) should
+        /// be included in the payload. This should only be used in debug scenarios.
+        /// </param>
+        /// <returns>Task which represents the pending write operation.</returns>
+        internal override Task WriteInStreamErrorAsync(ODataError error, bool includeDebugInformation)

```

Comment:
>WriteInStreamErrorAsync [](start = 31, length = 23)

Why do we have this async method on the Xml Metadata Output context but not the Json metadata output context?

Reply:
It's weird that we have such method in Metadata outputcontext, And there's no way to call this throw metadata output context.
I think we should remove this method but avoid changing existing class, i keep it unchanged.

So, For JSON metadata, i didn't add this method.

---
In reply to: [561201263](https://github.com/OData/odata.net/pull/1966#discussion_r561201263) [](ancestors = 561201263)